### ISSUE TITLE
Internal Error when upgrading from 0.41.0 to 0.42.1

### ISSUE THREAD

**Issue by NikolayMetchev** [pinterest/ktlint#1210]
After upgrading 2 files cause ktlint to fail with an Internal Error (indent)
[SettlementsSpec.kt.zip](https://github.com/pinterest/ktlint/files/6966917/SettlementsSpec.kt.zip)
[generators.kt.zip](https://github.com/pinterest/ktlint/files/6966925/generators.kt.zip)

## Expected Behavior
the file should get formatted

## Observed Behavior
SettlementsSpec.kt:0:0: Internal Error (indent). Please create a ticket at https://github.com/pinterest/ktlint/issues (if possible, provide the source code that triggered an error) (cannot be auto-corrected)

## Steps to Reproduce
Format the attached file 

## Your Environment
* Version of ktlint used: 0.42.1
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task):
Gradle Task:
task ktlintFmt(type: JavaExec) {
    mainClass.set("com.pinterest.ktlint.Main")
    classpath = configurations.ktlint
    args "-F", "src/**/*.kt"
    onlyIf {
        System.env['CI'] == null
    }
}
* Version of Gradle used (if applicable): 7.1.1
* Operating System and version: MacOS Big Sur
* Link to your project (if it's a public repository):

**Comment by shashachu** [comment]
@NikolayMetchev are you able to test with the latest SNAPSHOT? I believe this commit fixes your issues https://github.com/pinterest/ktlint/pull/1203

**Comment by mhernand40** [comment]
FWIW, the following snippet also triggers an Internal Error in ktlint `0.42.1`
```Kotlin
package com.repro

import kotlin.properties.Delegates

class Repro {
    private var repro
        by Delegates.vetoable("") { _, old, new ->
            true
        }
}
```

The error message is:
```
Internal Error (indent). Please create a ticket at https://github.com/pinterest/ktlint/issues (if possible, provide the source code that triggered an error) (cannot be auto-corrected)
```

No issue if I re-format to
```Kotlin
package com.repro

import kotlin.properties.Delegates

class Repro {
    private var repro by Delegates.vetoable("") { _, old, new ->
        true
    }
}
```

**Comment by paul-dingemans** [comment]
@mhernand40 Tnx for your example and observation how reformat made a difference. The stacktrace of the error is as follows:
```
[DEBUG] Internal Error (indent)
com.pinterest.ktlint.core.RuleExecutionException: java.lang.IllegalArgumentException: Count 'n' must be non-negative, but was -4.
	at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:406)
	at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:384)
	at com.pinterest.ktlint.core.KtLint$visitor$2.invoke(KtLint.kt:320)
	at com.pinterest.ktlint.core.KtLint$visitor$2.invoke(KtLint.kt:302)
	at com.pinterest.ktlint.core.KtLint.format(KtLint.kt:384)
	at com.pinterest.ktlint.internal.FileUtilsKt.formatFile(FileUtils.kt:202)
	at com.pinterest.ktlint.KtlintCommandLine.process(Main.kt:366)
	at com.pinterest.ktlint.KtlintCommandLine.access$process(Main.kt:89)
	at com.pinterest.ktlint.KtlintCommandLine$lintFiles$3.invoke$lambda-0(Main.kt:289)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: Count 'n' must be non-negative, but was -4.
	at kotlin.text.StringsKt__StringsJVMKt.repeat(StringsJVM.kt:790)
	at com.pinterest.ktlint.ruleset.standard.IndentationRule.visitWhiteSpace(IndentationRule.kt:1085)
	at com.pinterest.ktlint.ruleset.standard.IndentationRule.access$visitWhiteSpace(IndentationRule.kt:103)
	at com.pinterest.ktlint.ruleset.standard.IndentationRule$indent$2.invoke(IndentationRule.kt:600)
	at com.pinterest.ktlint.ruleset.standard.IndentationRule$indent$2.invoke(IndentationRule.kt:463)
	at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:234)
	at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:235)
	at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:235)
	at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:235)
	at com.pinterest.ktlint.ruleset.standard.IndentationRule.indent(IndentationRule.kt:462)
	at com.pinterest.ktlint.ruleset.standard.IndentationRule.visit(IndentationRule.kt:162)
	at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:391)
	... 13 more
	```

@t-kameyama Could you please have a look at this? This bug is somewhat related to your commit https://github.com/pinterest/ktlint/commit/aa47c9eba84b4b2b71ba4615817ef052ae8988b3. Problem is that the RPAR leads to an unwanted decrement of the expectedIndent cause its value to go below zero.

**Comment by BraisGabin** [comment]
I found other way to reproduce this issue:

```kotlin
object ApplicationComponentFactory : ApplicationComponent.Factory
by DaggerApplicationComponent.factory()
```

Testing with 0.43.1

if I reformat the code like this ktlint is happy and without errors:

```kotlin
object ApplicationComponentFactory :
    ApplicationComponent.Factory by DaggerApplicationComponent.factory()
```

**Comment by bbrinx** [comment]
Is there any update on this? Running into the same issue with the latest version 0.43.2

**Comment by paul-dingemans** [comment]
@BraisGabin I really love it when somebody has such a short example to reproduce an issue ;-) I could reproduce and solve it. 

Code sample:
```
object ApplicationComponentFactory : ApplicationComponent.Factory
by DaggerApplicationComponent.factory()
```

results in following stack trace:
```
Caused by: java.lang.IllegalArgumentException: Count 'n' must be non-negative, but was -4.
	at kotlin.text.StringsKt__StringsJVMKt.repeat(StringsJVM.kt:795)
	at com.pinterest.ktlint.ruleset.standard.IndentationRule.visitWhiteSpace(IndentationRule.kt:1049)
	at com.pinterest.ktlint.ruleset.standard.IndentationRule.access$visitWhiteSpace(IndentationRule.kt:104)
	at com.pinterest.ktlint.ruleset.standard.IndentationRule$indent$2.invoke(IndentationRule.kt:637)
	at com.pinterest.ktlint.ruleset.standard.IndentationRule$indent$2.invoke(IndentationRule.kt:499)
	at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:234)
	at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:235)
	at com.pinterest.ktlint.ruleset.standard.IndentationRule.indent(IndentationRule.kt:499)
	at com.pinterest.ktlint.ruleset.standard.IndentationRule.visit(IndentationRule.kt:163)
	at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:389)
	... 12 more
```
